### DISCUSSION TITLE
Need help with Vista 20se

### DISCUSSION THREAD

**Discussion by colednichols** [Dilbert66/esphome-vistaECP#81] [General]
I successfully configured the yaml and flashed the board, but I can't seem to read any data from the panel. The log prints this repeatedly, cycling through zones:
`'<zone name>': Sending state 'C'`

I have 1 physical alpha-numeric keypad and the manual says to ensure additional keypads are unaddressed by using address 31, which I have tried with no avail. The log matches what I saw when it was not connected at all.

I can't find much information involving the Vista 20se and I was wondering if anyone could help me get it working.

**Comment by Dilbert66** [comment]
You didnt indicate which code are using.  There is a special branch for the SE variant:

https://github.com/Dilbert66/esphome-vistaECP/tree/master/src/vista20SE

**Reply by colednichols** [reply]
Thank you and sorry for the ignorance. Not sure how I completely overlooked something so obvious. Although now that I've done that, I am running into another problem. Here is the log when installing the yaml:

In file included from src/main.cpp:67:0:
src/vistaAlarm.h:255:5: error: 'alarmStatus' does not name a type
     alarmStatus fireStatus,
     ^
src/vistaAlarm.h: In member function 'virtual void esphome::vistaECPHome::update()':
src/vistaAlarm.h:758:17: error: 'fireStatus' was not declared in this scope
                 fireStatus.zone = vista.statusFlags.zone;
                 ^
src/vistaAlarm.h:771:21: error: 'alarmStatus' was not declared in this scope
                     alarmStatus.zone = vista.statusFlags.zone;
                     ^
src/vistaAlarm.h:775:21: error: 'panicStatus' was not declared in this scope
                     panicStatus.zone = vista.statusFlags.zone;
                     ^
src/vistaAlarm.h:830:17: error: 'alarmStatus' was not declared in this scope
                 alarmStatus.zone = 99;
                 ^
src/vistaAlarm.h:860:29: error: 'fireStatus' was not declared in this scope
             if ((millis() - fireStatus.time) > TTL) fireStatus.state = false;
                             ^
src/vistaAlarm.h:861:29: error: 'alarmStatus' was not declared in this scope
             if ((millis() - alarmStatus.time) > TTL) alarmStatus.state = false;             
                             ^
src/vistaAlarm.h:862:29: error: 'panicStatus' was not declared in this scope
             if ((millis() - panicStatus.time) > TTL) panicStatus.state = false;
                             ^
src/vistaAlarm.h:869:37: error: 'alarmStatus' was not declared in this scope
             currentLightState.alarm=alarmStatus.state;
                                     ^
*** [/data/vistaalarm/.pioenvs/vistaalarm/src/main.cpp.o] Error 1

**Comment by Dilbert66** [comment]
Ah, yes. there is a bug in the master version. Try the "dev" branch version instead.

**Reply by colednichols** [reply]
Thank you so much. I was able to get it running through the "dev" branch, though I did find 2 issues (I think they are issues as I could not compile before changing them):

Firstly, on line 329 of VistaAlarm_single_partition_text_zone_sensors.yaml, the following block is defined twice:
```
  - platform: template
    id: rf1
    name: "$systemName RF Msg"
    icon: "mdi:alert-box"
```

Secondly, on line 457 of vistaAlarm.h, I found that it would not compile unless I changed `vista.write(keys,addr);` to `vista.write(keys);`.

**Comment by Dilbert66** [comment]
Ah, ok. Tks. Yes, those were cut and paste errors on my part.  I had not fully retested that code after doing some multi partition updates.  I've corrected the files and pushed.  I've also corrected the vistaalarm.h version in branch master.

Edit: Just noticed that the dev version of the SE code is not correct. I was doing some updates on it and realize that i left it incomplete.   Use the version from master for the vista20se and not dev.     I'm working on multi partition support and was starting to add it to the SE but never completed.

**Reply by colednichols** [reply]
Sounds good, thanks!
